### ISSUE TITLE
Type error with permissionOverwrites

### ISSUE THREAD

**Issue by Fyrlex** [discordjs/discord.js#10450]
### Which package is this bug report for?

discord.js

### Issue description

With an existing guild text channel, an error is thrown when setting the `permissionOverwrites` with the required types. 

Here is an example

```ts
const channel = await client.channels.fetch('some ID that exists as a text channel')

channel.permissionOverwrites
            .set([
              {
                id: client.user.id,
                allow: [
                  'SendMessages',
                ],
              },
            ]);
```
An error will be thrown
```
TypeError [InvalidType]: Supplied parameter is not a User nor a Role.
```

And the typings for this method I am using have `id` as a `Snowflake`
![image](https://github.com/user-attachments/assets/912b3fa9-283b-4101-b4f3-a39bde9491a4)

When using the `client.user` instead, there is no issue and the permissions are correctly changed.

### Versions

- discord.js 14.15.3
- typescript 5.4.5
- node 20.13.1

### Issue priority

Low (slightly annoying)

### Which partials do you have configured?

Not applicable

### Which gateway intents are you subscribing to?

Not applicable

**Comment by Renegade334** [comment]
If you supply a snowflake as the id of an OverwriteData object and don't specify a type, then the user/role needs to be cached (otherwise d.js can't work out whether to set a role overwrite or a member overwrite).

If you don't know that the user/role is cached from context, then you can either pass the User/Role object directly as you mentioned, or supply a value for `type`.

```js
channel.permissionOverwrites
            .set([
              {
                id: client.user.id,
                type: 'member',
                allow: [
                  'SendMessages',
                ],
              },
            ]);

**Comment by Jiralite** [comment]
That's correct. I believe it would be a good idea to adjust the error message here as it is misleading (the supplied parameter was indeed a role or a user). It should ideally mention that it was unable to resolve the type.

**Comment by Renegade334** [comment]
I suppose the alternative (albeit breaking) approach would be to switch to using `resolveId` rather than `resolve`, and enforce `overwrite.type` as mandatory if `overwrite.id` is a string, to remove the need for cache lookups and thereby avoid cache-dependent error behaviour?

**Comment by Waheedsys** [comment]
can i work on this.

**Comment by didinele** [comment]
Yes. If you know how to, feel free to open a pull request.

**Comment by Waheedsys** [comment]
sure, thank you @didinele

**Comment by baraich** [comment]
@Waheedsys Hi, are you skilled to work the issue, if not let me take over. Thanks.

**Comment by Waheedsys** [comment]
Hey @Jiralite, could you please check if the ESLint error has been resolved? Thanks!

**Comment by KingsMMA** [comment]
I agree that the existing error message is useless, I've been trying to debug how an invalid ID somehow ended up in the request until I found this issue.
